daemon/config/config.go

```diff
@@ -139,10 +159,11 @@ type TLSOptions struct {
 
 // DNSConfig defines the DNS configurations.
 type DNSConfig struct {
-	DNS           []net.IP `json:"dns,omitempty"`
-	DNSOptions    []string `json:"dns-opts,omitempty"`
-	DNSSearch     []string `json:"dns-search,omitempty"`
-	HostGatewayIP net.IP   `json:"host-gateway-ip,omitempty"`
+	DNS            []net.IP `json:"dns,omitempty"`
+	DNSOptions     []string `json:"dns-opts,omitempty"`
+	DNSSearch      []string `json:"dns-search,omitempty"`
+	HostGatewayIP  net.IP   `json:"host-gateway-ip,omitempty"` // Deprecated: this single-IP is migrated to HostGatewayIPs
```

Comment:
Looks like this would require it to be a pointer for the `omitempty` to work (also see my example below; https://go.dev/play/p/m-9f_mVwuGW)

Reply:
The only change here is to deprecate the existing field.

I don't think this struct is ever marshalled - just unmarshalled from the config file ... so I'm not sure why any of the fields have `omitempty`.

Reply:
Yeah, not super critical probably; I think we use the JSON format to log it when reloading the config; perhaps I should try if that would work with structured logs (although it can be useful to have the log log the config in the format used for `config.json` 🤔 https://github.com/moby/moby/blob/803534fb4e784d3e6bef9846d64ce4960ad1ec80/daemon/reload.go#L118-L129